internal/controllers/clusterrequest_controller.go

```diff
@@ -705,6 +716,88 @@ func (t *clusterRequestReconcilerTask) handleClusterInstallation(ctx context.Con
 	return nil
 }
 
+// handleClusterPolicyConfiguration updates the ClusterRequest status to reflect the status
+// of the policies that match the managed cluster created through the ClusterRequest.
+func (t *clusterRequestReconcilerTask) handleClusterPolicyConfiguration(ctx context.Context) error {
+	if t.object.Status.ClusterInstanceRef == nil {
+		return fmt.Errorf("status.clusterInstanceRef is empty")
+	}
+	// Get all the child policies in the namespace of the managed cluster created through
+	// the ClusterRequest.
+	policies := &policiesv1.PolicyList{}
+	listOpts := []client.ListOption{
+		client.HasLabels{utils.ChildPolicyRootPolicyLabel},
+		client.MatchingLabels(
+			map[string]string{
+				utils.ChildPolicyClusterNameLabel:      t.object.Status.ClusterInstanceRef.Name,
+				utils.ChildPolicyClusterNamespaceLabel: t.object.Status.ClusterInstanceRef.Name,
+			},
+		),
+	}
+
+	err := t.client.List(ctx, policies, listOpts...)
+	if err != nil {
+		return fmt.Errorf("failed to list Policies: %w", err)
+	}
+
+	allPoliciesCompliant := true
+	nonCompliantPolicyInEnforce := false
+	var targetPolicies []oranv1alpha1.PolicyDetails
+	// Go through all the policies and get those that are matched with the managed cluster created
+	// by the current cluster request.
+	for _, policy := range policies.Items {
+		if policy.Status.ComplianceState == policiesv1.NonCompliant {
+			allPoliciesCompliant = false
+			if policy.Spec.RemediationAction == policiesv1.Enforce || policy.Spec.RemediationAction == "enforce" {
```

Comment:
what about `strings.EqualFold(policy.Spec.RemediationAction, string(policiesv1.Enforce))`? This check is case-insensitive